packages/db/src/lib/QuerySearchClient.ts

```diff
@@ -29,12 +32,31 @@ class QuerySearchClient<RecordType extends BaseRecord> {
    * @throws
    */
   private async fetchRecords() {
-    this.records = await (
-      this.query
-        .offset(this.offset)
-        .limit(this.limit)
+    return await pRetry(
```

Comment:
I think we should have a common method, like
RetryOnDbConnectionTerminateError(methodName, options) {
  await pRetry(methodName, { retries: 3, onFailed , ...options})
}

Reply:
https://github.com/nasa/cumulus/pull/3332/commits/afb3b702976a0e4db8c5e63f6520c39e423ec898#diff-4407d17441d12af663b8a19415c93087978d8d8a6a80b3e7ec4ea9222890c388R8